unsigned int tomo::IoLoop::epoll_pass<false, true, true>(int)
--- PRE

+++ POST

@@ -2,7 +2,7 @@

 0004 [ 2] push %r15
 0006 [ 2] push %r14
 0008 [ 2] push %r13
-000a [ 3] mov %rdi,%r14
+000a [ 3] mov %rdi,%r15
 000d [ 2] push %r12
 000f [ 1] push %rbp
 0010 [ 1] push %rbx
@@ -13,202 +13,195 @@

 0024 [ 5] mov $0x100,%edx
 0029 [ 3] mov %rbx,%rsi
 002c [ 3] xor %r12d,%r12d
-002f [ 6] mov $0x0,%r15d
+002f [ 6] mov $0x0,%r14d
 0035 [ 9] mov %fs:0x28,%rax
 003e [ 5] mov %rax,0x28(%rsp)
 0043 [ 2] xor %eax,%eax
 0045 [ 5] call <epoll_wait@plt>
 004a [ 2] test %eax,%eax
 004c [ 4] cmovns %eax,%r12d
-0050 [ 7] mov %r12d,0xc18(%r14)
-0057 [ 6] jle <+0x13f>
-005d [ 7] mov 0x190(%r14),%rax
-0064 [ 3] movslq %r12d,%rdx
-0067 [ 2] xor %ebp,%ebp
-0069 [ 7] lea <rip>(%rip),%r13 # <typeinfo name for std::_Sp_counted_ptr_inplace<tomo::AofGroupDecision, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>+0x79c>
-0070 [ 7] add %rdx,0x520(%rax)
-0077 [ 7] lea 0xac8(%r14),%rax
-007e [ 5] mov %rax,0x8(%rsp)
-0083 [ 5] lea 0x20(%rsp),%rax
-0088 [ 4] mov %rax,(%rsp)
-008c [ 4] nopl 0x0(%rax)
-0090 [ 4] mov 0x4(%rbx),%rdx
-0094 [ 3] mov %rdx,%rax
+0050 [ 7] mov %r12d,0xc18(%r15)
+0057 [ 2] jle <+0xcb>
+0059 [ 7] mov 0x190(%r15),%rax
+0060 [ 3] movslq %r12d,%rdx
+0063 [ 2] xor %ebp,%ebp
+0065 [ 7] lea <rip>(%rip),%r13 # <typeinfo name for std::_Sp_counted_ptr_inplace<tomo::AofGroupDecision, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>+0x79c>
+006c [ 7] add %rdx,0x520(%rax)
+0073 [ 7] lea 0xac8(%r15),%rax
+007a [ 5] mov %rax,0x8(%rsp)
+007f [ 5] lea 0x20(%rsp),%rax
+0084 [ 4] mov %rax,(%rsp)
+0088 [ 8] nopl 0x0(%rax,%rax,1)
+0090 [ 4] mov 0x4(%rbx),%rsi
+0094 [ 3] mov %rsi,%rax
 0097 [ 4] shr $0x38,%rax
 009b [ 2] cmp $0x11,%al
-009d [ 6] ja <+0x130>
-00a3 [ 5] movslq 0x0(%r13,%rax,4),%rax
-00a8 [ 3] add %r13,%rax
-00ab [ 3] notrack jmp *%rax
-00ae [ 2] mov (%rbx),%eax
-00b0 [ 5] test $0x2019,%eax
-00b5 [ 2] je <+0xbb>
-00b7 [ 4] movb $0x0,0x29(%rdx)
-00bb [ 2] test $0x1c,%al
-00bd [ 2] je <+0xe9>
-00bf [ 4] cmpb $0x0,0x30(%rdx)
-00c3 [ 5] mov %rdx,0x20(%rsp)
-00c8 [ 2] jne <+0xe9>
-00ca [ 5] lea 0x20(%rsp),%rsi
-00cf [ 7] lea 0x198(%r14),%rdi
-00d6 [ 4] movb $0x1,0x30(%rdx)
-00da [ 5] mov %rdx,0x10(%rsp)
-00df [ 5] call <std::deque<tomo::Client*, std::allocator<tomo::Client*> >::push_back(tomo::Client* const&)>
-00e4 [ 5] mov 0x10(%rsp),%rdx
-00e9 [ 3] mov %rdx,%rsi
-00ec [ 3] mov %r14,%rdi
-00ef [ 5] mov %rdx,0x10(%rsp)
-00f4 [ 5] call <tomo::IoLoop::tls_slot_conn(tomo::Client const*)>
-00f9 [ 5] mov 0x10(%rsp),%rdx
-00fe [ 3] test %rax,%rax
-0101 [ 2] je <+0x10d>
-0103 [ 6] movw $0x0,0x3e(%rax)
-0109 [ 4] movb $0x0,0x29(%rdx)
-010d [ 3] mov %rdx,%rsi
-0110 [ 3] mov %r14,%rdi
-0113 [ 5] call <tomo::IoLoop::mark_active(tomo::Client*)>
-0118 [ 8] nopl 0x0(%rax,%rax,1)
-0120 [ 3] inc %r15d
-0123 [11] data16 cs nopw 0x0(%rax,%rax,1)
-012e [ 2] xchg %ax,%ax
-0130 [ 2] inc %ebp
-0132 [ 4] add $0xc,%rbx
-0136 [ 3] cmp %r12d,%ebp
-0139 [ 6] jl <+0x90>
-013f [ 5] mov 0x28(%rsp),%rax
-0144 [ 9] sub %fs:0x28,%rax
-014d [ 6] jne <+0x3a8>
-0153 [ 4] add $0x38,%rsp
-0157 [ 3] mov %r15d,%eax
-015a [ 1] pop %rbx
-015b [ 1] pop %rbp
-015c [ 2] pop %r12
-015e [ 2] pop %r13
-0160 [ 2] pop %r14
-0162 [ 2] pop %r15
-0164 [ 1] ret
-0165 [11] data16 cs nopw 0x0(%rax,%rax,1)
-0170 [ 9] movq $0x0,0x20(%rsp)
-0179 [ 7] nopl 0x0(%rax)
-0180 [ 7] mov 0xbc8(%r14),%edi
-0187 [ 4] mov (%rsp),%rsi
-018b [ 5] mov $0x8,%edx
-0190 [ 5] call <read@plt>
-0195 [ 4] cmp $0x8,%rax
-0199 [ 2] je <+0x180>
-019b [ 7] mov 0x190(%r14),%rax
-01a2 [ 2] inc %ebp
-01a4 [ 3] inc %r15d
-01a7 [ 4] add $0xc,%rbx
-01ab [ 7] incq 0x440(%rax)
-01b2 [ 3] cmp %r12d,%ebp
-01b5 [ 6] jl <+0x90>
-01bb [ 2] jmp <+0x13f>
-01bd [ 3] nopl (%rax)
-01c0 [ 7] mov 0x190(%r14),%rax
-01c7 [ 7] mov 0x188(%r14),%rdi
-01ce [ 5] mov 0x8(%rsp),%rdx
-01d3 [ 2] inc %ebp
-01d5 [ 3] inc %r15d
-01d8 [ 4] add $0xc,%rbx
-01dc [ 2] mov (%rax),%esi
-01de [ 5] call <tomo::DatabaseMap::consume_wake(unsigned int, tomo::Ring&)>
-01e3 [ 3] cmp %r12d,%ebp
-01e6 [ 6] jl <+0x90>
-01ec [ 5] jmp <+0x13f>
-01f1 [11] data16 cs nopw 0x0(%rax,%rax,1)
-01fc [ 4] nopl 0x0(%rax)
-0200 [ 7] mov 0x188(%r14),%rax
-0207 [ 7] movzbl 0x115e0(%rax),%eax
-020e [ 2] cmp $0x3,%al
-0210 [ 6] ja <+0x130>
-0216 [ 7] mov 0xaa4(%r14),%eax
-021d [ 4] mov %eax,0x10(%rsp)
-0221 [ 2] test %eax,%eax
-0223 [ 6] js <+0x130>
-0229 [ 8] movl $0x0,0x1c(%rsp)
-0231 [ 4] mov 0x10(%rsp),%edi
-0235 [ 2] xor %edx,%edx
-0237 [ 2] xor %esi,%esi
-0239 [ 5] mov $0x80800,%ecx
-023e [ 5] call <accept4@plt>
-0243 [ 2] test %eax,%eax
-0245 [ 6] jns <+0x390>
-024b [ 5] call <__errno_location@plt>
-0250 [ 2] mov (%rax),%eax
-0252 [ 3] cmp $0x4,%eax
-0255 [ 6] jne <+0x333>
-025b [ 2] jmp <+0x231>
-025d [ 3] nopl (%rax)
-0260 [10] movabs $0xffffffffffff,%rax
-026a [ 3] and %rax,%rdx
-026d [ 6] je <+0x130>
-0273 [ 6] mov 0x788(%rdx),%eax
-0279 [ 7] mov 0x190(%r14),%rcx
-0280 [ 2] cmp %eax,(%rcx)
-0282 [ 6] jne <+0x130>
-0288 [ 4] cmpb $0x0,0x33(%rdx)
-028c [ 6] jne <+0x130>
-0292 [ 6] mov 0x784(%rdx),%eax
-0298 [ 3] cmp $0xfffffffe,%eax
-029b [ 6] je <+0x130>
-02a1 [ 7] mov 0x288(%r14),%rax
-02a8 [ 7] mov 0x290(%r14),%rcx
-02af [ 3] cmp %rcx,%rax
-02b2 [ 2] jne <+0x2cd>
-02b4 [ 5] jmp <+0xae>
-02b9 [ 7] nopl 0x0(%rax)
-02c0 [ 4] add $0x28,%rax
-02c4 [ 3] cmp %rax,%rcx
-02c7 [ 6] je <+0xae>
-02cd [ 3] cmp (%rax),%rdx
-02d0 [ 2] jne <+0x2c0>
-02d2 [ 5] jmp <+0x130>
-02d7 [ 9] nopw 0x0(%rax,%rax,1)
-02e0 [ 7] mov 0x188(%r14),%rax
-02e7 [ 7] movzbl 0x115e0(%rax),%eax
-02ee [ 2] cmp $0x3,%al
-02f0 [ 6] ja <+0x130>
-02f6 [ 7] mov 0xaa0(%r14),%eax
-02fd [ 4] mov %eax,0x10(%rsp)
-0301 [ 2] test %eax,%eax
-0303 [ 6] js <+0x130>
-0309 [ 8] movl $0x0,0x1c(%rsp)
-0311 [ 4] mov 0x10(%rsp),%edi
-0315 [ 2] xor %edx,%edx
-0317 [ 2] xor %esi,%esi
-0319 [ 5] mov $0x80800,%ecx
-031e [ 5] call <accept4@plt>
-0323 [ 2] test %eax,%eax
-0325 [ 2] jns <+0x370>
-0327 [ 5] call <__errno_location@plt>
-032c [ 2] mov (%rax),%eax
-032e [ 3] cmp $0x4,%eax
-0331 [ 2] je <+0x311>
-0333 [ 3] cmp $0xb,%eax
-0336 [ 2] je <+0x346>
-0338 [ 7] mov 0x190(%r14),%rax
-033f [ 7] incq 0x458(%rax)
-0346 [ 4] mov 0x1c(%rsp),%eax
-034a [ 2] inc %ebp
-034c [ 4] add $0xc,%rbx
-0350 [ 3] add %eax,%r15d
-0353 [ 3] cmp %r12d,%ebp
-0356 [ 6] jl <+0x90>
-035c [ 5] jmp <+0x13f>
-0361 [11] data16 cs nopw 0x0(%rax,%rax,1)
-036c [ 4] nopl 0x0(%rax)
-0370 [ 5] mov $0x1,%edx
-0375 [ 2] mov %eax,%esi
-0377 [ 3] mov %r14,%rdi
-037a [ 4] incl 0x1c(%rsp)
-037e [ 5] call <void tomo::IoLoop::admit_fd<true, true>(int, tomo::UrKind)>
-0383 [ 2] jmp <+0x311>
-0385 [11] data16 cs nopw 0x0(%rax,%rax,1)
-0390 [ 5] mov $0x9,%edx
-0395 [ 2] mov %eax,%esi
-0397 [ 3] mov %r14,%rdi
-039a [ 4] incl 0x1c(%rsp)
-039e [ 5] call <void tomo::IoLoop::admit_fd<true, true>(int, tomo::UrKind)>
-03a3 [ 5] jmp <+0x231>
-03a8 [ 5] call <__stack_chk_fail@plt>
+009d [ 2] ja <+0xb0>
+009f [ 5] movslq 0x0(%r13,%rax,4),%rax
+00a4 [ 3] add %r13,%rax
+00a7 [ 3] notrack jmp *%rax
+00aa [ 6] nopw 0x0(%rax,%rax,1)
+00b0 [ 3] mov %r15,%rdi
+00b3 [ 5] call <tomo::IoLoop::fliplisten_epoll(unsigned long)>
+00b8 [ 3] add %eax,%r14d
+00bb [ 5] nopl 0x0(%rax,%rax,1)
+00c0 [ 2] inc %ebp
+00c2 [ 4] add $0xc,%rbx
+00c6 [ 3] cmp %r12d,%ebp
+00c9 [ 2] jl <+0x90>
+00cb [ 5] mov 0x28(%rsp),%rax
+00d0 [ 9] sub %fs:0x28,%rax
+00d9 [ 6] jne <+0x378>
+00df [ 4] add $0x38,%rsp
+00e3 [ 3] mov %r14d,%eax
+00e6 [ 1] pop %rbx
+00e7 [ 1] pop %rbp
+00e8 [ 2] pop %r12
+00ea [ 2] pop %r13
+00ec [ 2] pop %r14
+00ee [ 2] pop %r15
+00f0 [ 1] ret
+00f1 [11] data16 cs nopw 0x0(%rax,%rax,1)
+00fc [ 4] nopl 0x0(%rax)
+0100 [10] movabs $0xffffffffffff,%rdx
+010a [ 3] and %rsi,%rdx
+010d [ 2] je <+0xc0>
+010f [ 6] mov 0x788(%rdx),%eax
+0115 [ 7] mov 0x190(%r15),%rcx
+011c [ 2] cmp %eax,(%rcx)
+011e [ 2] jne <+0xc0>
+0120 [ 4] cmpb $0x0,0x33(%rdx)
+0124 [ 2] jne <+0xc0>
+0126 [ 6] mov 0x784(%rdx),%eax
+012c [ 3] cmp $0xfffffffe,%eax
+012f [ 2] je <+0xc0>
+0131 [ 7] mov 0x288(%r15),%rax
+0138 [ 7] mov 0x290(%r15),%rcx
+013f [ 3] cmp %rcx,%rax
+0142 [ 2] je <+0x162>
+0144 [11] data16 cs nopw 0x0(%rax,%rax,1)
+014f [ 1] nop
+0150 [ 3] cmp (%rax),%rdx
+0153 [ 6] je <+0xc0>
+0159 [ 4] add $0x28,%rax
+015d [ 3] cmp %rax,%rcx
+0160 [ 2] jne <+0x150>
+0162 [ 2] mov (%rbx),%eax
+0164 [ 5] test $0x2019,%eax
+0169 [ 2] je <+0x16f>
+016b [ 4] movb $0x0,0x29(%rdx)
+016f [ 2] test $0x1c,%al
+0171 [ 2] je <+0x19d>
+0173 [ 4] cmpb $0x0,0x30(%rdx)
+0177 [ 5] mov %rdx,0x20(%rsp)
+017c [ 2] jne <+0x19d>
+017e [ 5] lea 0x20(%rsp),%rsi
+0183 [ 7] lea 0x198(%r15),%rdi
+018a [ 4] movb $0x1,0x30(%rdx)
+018e [ 5] mov %rdx,0x10(%rsp)
+0193 [ 5] call <std::deque<tomo::Client*, std::allocator<tomo::Client*> >::push_back(tomo::Client* const&)>
+0198 [ 5] mov 0x10(%rsp),%rdx
+019d [ 3] mov %rdx,%rsi
+01a0 [ 3] mov %r15,%rdi
+01a3 [ 5] mov %rdx,0x10(%rsp)
+01a8 [ 5] call <tomo::IoLoop::tls_slot_conn(tomo::Client const*)>
+01ad [ 5] mov 0x10(%rsp),%rdx
+01b2 [ 3] test %rax,%rax
+01b5 [ 2] je <+0x1c1>
+01b7 [ 6] movw $0x0,0x3e(%rax)
+01bd [ 4] movb $0x0,0x29(%rdx)
+01c1 [ 3] mov %rdx,%rsi
+01c4 [ 3] mov %r15,%rdi
+01c7 [ 5] call <tomo::IoLoop::mark_active(tomo::Client*)>
+01cc [ 4] nopl 0x0(%rax)
+01d0 [ 3] inc %r14d
+01d3 [ 5] jmp <+0xc0>
+01d8 [ 8] nopl 0x0(%rax,%rax,1)
+01e0 [ 9] movq $0x0,0x20(%rsp)
+01e9 [ 7] nopl 0x0(%rax)
+01f0 [ 7] mov 0xbc8(%r15),%edi
+01f7 [ 4] mov (%rsp),%rsi
+01fb [ 5] mov $0x8,%edx
+0200 [ 5] call <read@plt>
+0205 [ 4] cmp $0x8,%rax
+0209 [ 2] je <+0x1f0>
+020b [ 7] mov 0x190(%r15),%rax
+0212 [ 3] inc %r14d
+0215 [ 7] incq 0x440(%rax)
+021c [ 5] jmp <+0xc0>
+0221 [11] data16 cs nopw 0x0(%rax,%rax,1)
+022c [ 4] nopl 0x0(%rax)
+0230 [ 7] mov 0x188(%r15),%rax
+0237 [ 7] movzbl 0x115e0(%rax),%eax
+023e [ 2] cmp $0x3,%al
+0240 [ 6] ja <+0xc0>
+0246 [ 7] mov 0xaa0(%r15),%eax
+024d [ 4] mov %eax,0x10(%rsp)
+0251 [ 2] test %eax,%eax
+0253 [ 6] js <+0xc0>
+0259 [ 8] movl $0x0,0x1c(%rsp)
+0261 [ 4] mov 0x10(%rsp),%edi
+0265 [ 2] xor %edx,%edx
+0267 [ 2] xor %esi,%esi
+0269 [ 5] mov $0x80800,%ecx
+026e [ 5] call <accept4@plt>
+0273 [ 2] test %eax,%eax
+0275 [ 6] jns <+0x360>
+027b [ 5] call <__errno_location@plt>
+0280 [ 2] mov (%rax),%eax
+0282 [ 3] cmp $0x4,%eax
+0285 [ 6] jne <+0x313>
+028b [ 2] jmp <+0x261>
+028d [ 3] nopl (%rax)
+0290 [ 7] mov 0x190(%r15),%rax
+0297 [ 7] mov 0x188(%r15),%rdi
+029e [ 5] mov 0x8(%rsp),%rdx
+02a3 [ 3] inc %r14d
+02a6 [ 2] mov (%rax),%esi
+02a8 [ 5] call <tomo::DatabaseMap::consume_wake(unsigned int, tomo::Ring&)>
+02ad [ 5] jmp <+0xc0>
+02b2 [11] data16 cs nopw 0x0(%rax,%rax,1)
+02bd [ 3] nopl (%rax)
+02c0 [ 7] mov 0x188(%r15),%rax
+02c7 [ 7] movzbl 0x115e0(%rax),%eax
+02ce [ 2] cmp $0x3,%al
+02d0 [ 6] ja <+0xc0>
+02d6 [ 7] mov 0xaa4(%r15),%eax
+02dd [ 4] mov %eax,0x10(%rsp)
+02e1 [ 2] test %eax,%eax
+02e3 [ 6] js <+0xc0>
+02e9 [ 8] movl $0x0,0x1c(%rsp)
+02f1 [ 4] mov 0x10(%rsp),%edi
+02f5 [ 2] xor %edx,%edx
+02f7 [ 2] xor %esi,%esi
+02f9 [ 5] mov $0x80800,%ecx
+02fe [ 5] call <accept4@plt>
+0303 [ 2] test %eax,%eax
+0305 [ 2] jns <+0x340>
+0307 [ 5] call <__errno_location@plt>
+030c [ 2] mov (%rax),%eax
+030e [ 3] cmp $0x4,%eax
+0311 [ 2] je <+0x2f1>
+0313 [ 3] cmp $0xb,%eax
+0316 [ 2] je <+0x326>
+0318 [ 7] mov 0x190(%r15),%rax
+031f [ 7] incq 0x458(%rax)
+0326 [ 4] mov 0x1c(%rsp),%eax
+032a [ 3] add %eax,%r14d
+032d [ 5] jmp <+0xc0>
+0332 [11] data16 cs nopw 0x0(%rax,%rax,1)
+033d [ 3] nopl (%rax)
+0340 [ 5] mov $0x9,%edx
+0345 [ 2] mov %eax,%esi
+0347 [ 3] mov %r15,%rdi
+034a [ 4] incl 0x1c(%rsp)
+034e [ 5] call <void tomo::IoLoop::admit_fd<true, true>(int, tomo::UrKind)>
+0353 [ 2] jmp <+0x2f1>
+0355 [11] data16 cs nopw 0x0(%rax,%rax,1)
+0360 [ 5] mov $0x1,%edx
+0365 [ 2] mov %eax,%esi
+0367 [ 3] mov %r15,%rdi
+036a [ 4] incl 0x1c(%rsp)
+036e [ 5] call <void tomo::IoLoop::admit_fd<true, true>(int, tomo::UrKind)>
+0373 [ 5] jmp <+0x261>
+0378 [ 5] call <__stack_chk_fail@plt>
